### PR TITLE
Fixes for new module

### PR DESCRIPTION
**Pull request by marekrozmus**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,9 +45,12 @@
   },
   "repository": {
     "type": "git",
-    "url": "https://github.com/osano/cookieconsent.git"
+    "url": "https://github.com/sandstreamdev/cookieconsent"
+  },
+  "author": {
```
Comment (przemyslawzalewski) [inline package.json]: Can we list multiple authors and point to the original repo as well?

File: src/index.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import "./styles/main.scss"
 import CookieConsent from "./models/CookieConsent"
 
 if (typeof exports !== 'undefined') {
-  module.exports = CookieConsent
+  exports.CookieConsent = CookieConsent
```
Comment (przemyslawzalewski) [inline src/index.js:8]: As imports are used, please use ESM `export default CookieConsent`.

**Review by przemyslawzalewski — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "cookieconsent",
-  "version": "4.0.0",
-  "description": "Osano cookie consent tool.",
+  "name": "@sandstreamdev/cookieconsent",
```
Comment (przemyslawzalewski) [inline package.json:2]: Please also change the entry in `package-lock.json` if it is there.

File: bower.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
```
Comment (przemyslawzalewski) [inline bower.json]: Maybe let's remove this file - who is ever using bower nowadays?

File: package.json (hunk 1 above)
Comment (marekrozmus) [reply package.json]: We could leave the old name and add the contributors link as it is right now.

About the repository link - it is link to our fork so if someone will open it there will be information about it

File: README.md [hunk 5]
```diff
@@ -1,3 +1,95 @@
...
+    categoryUncategorized: 'My new awesome text for uncategorized category'
+  }
+});
+```
+
+#### How to set the Google Tag Manager (GTM) cookie only if user accepted `Analytics` cookies?
+
+Check [GDPR example](./examples/example-gdpr.html) to see the solution.
+
+It handles also scenario when user revoke consent.
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
It also handles scenarios where users revoke their consent.
```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,95 @@
...
+    categoryAnalytics: 'My new awesome text for analytics category',
+    categoryEssential: 'My new awesome text for essential category',
+    categoryMarketing: 'My new awesome text for marketing category',
+    categoryPersonalization: 'My new awesome text for personalization category',
+    categoryUncategorized: 'My new awesome text for uncategorized category'
+  }
+});
+```
+
+#### How to set the Google Tag Manager (GTM) cookie only if user accepted `Analytics` cookies?
```
Comment (przemyslawzalewski) [inline README.md]: Please avoid using backticks and code blocks for non-code related words. Use emphasis (double **) or double underlines (or none if not really needed).

```suggestion
#### How to set the Google Tag Manager (GTM) cookie only if the user has accepted analytics cookies?
```

File: README.md [hunk 7]
```diff
@@ -1,3 +1,95 @@
...
+```
+const cc = new CC({
+  //...options,
+  consentSettingsElementId: 'btn-revokeChoice'
+});
+```
+
+#### Replace default `Read more` texts for categories
+
+All or some subset of categories' read more text can be changed.
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
All or the desired subset of categories' read more text can be changed.
```

File: README.md [hunk 8]
```diff
@@ -1,3 +1,95 @@
...
+As in GDPR you need to provide functionality to let the user change consents `consentSettingsElementId` was added. It is id of element on the page that will reopen categories dialog after clicking it.
+
+```
+const cc = new CC({
+  //...options,
+  consentSettingsElementId: 'btn-revokeChoice'
+});
+```
+
+#### Replace default `Read more` texts for categories
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
#### How to replace the default `Read more` texts for categories?
```

File: README.md [hunk 9]
```diff
@@ -1,3 +1,95 @@
...
+});
+```
+
+#### How to set the Google Tag Manager (GTM) cookie only if user accepted `Analytics` cookies?
+
+Check [GDPR example](./examples/example-gdpr.html) to see the solution.
+
+It handles also scenario when user revoke consent.
+
+#### Can be styled fully (or almost ;)) with css
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
#### Styling with CSS
```

File: README.md [hunk 10]
```diff
@@ -1,3 +1,95 @@
...
+
+Styled version used in one of our projects:
+
+<p align="center"><img width="800" src="images/cc-dialog-styled.png"></a></p>
+
+<p align="center"><img width="800" src="images/categories-dialog-styled.png"></a></p>
+
+## And that's all from [Sandstream Development](https://github.com/sandstreamdev). Enjoy!
```
Comment (przemyslawzalewski) [inline README.md]: I think this should not be heading :)

File: README.md [hunk 11]
```diff
@@ -1,3 +1,95 @@
...
+
+Check [GDPR example](./examples/example-gdpr.html) to see the solution.
+
+It handles also scenario when user revoke consent.
+
+#### Can be styled fully (or almost ;)) with css
+
+Styled version used in one of our projects:
+
+<p align="center"><img width="800" src="images/cc-dialog-styled.png"></a></p>
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
<p align="center"><img width="800" src="./images/cc-dialog-styled.png"></a></p>
```

File: README.md [hunk 12]
```diff
@@ -1,3 +1,95 @@
...
+
+<p align="center"><img width="800" src="images/categories-dialog-styled.png"></a></p>
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
<p align="center"><img width="800" src="./images/categories-dialog-styled.png"></a></p>
```

File: README.md [hunk 13]
```diff
@@ -1,3 +1,95 @@
...
+  content: {
+    privacyPolicyLink: '/privacy-policy',
+    cookiePolicyLink: '/cookie-policy'
+  }
+})
+```
+
+#### Consent settings
+
+As in GDPR you need to provide functionality to let the user change consents `consentSettingsElementId` was added. It is id of element on the page that will reopen categories dialog after clicking it.
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
As in GDPR, you need to provide the functionality to let the user change consents, so the `consentSettingsElementId` was added. It is the id of the element on the page that will reopen the categories dialog after clicking it.
```

File: README.md [hunk 14]
```diff
@@ -1,3 +1,95 @@
...
+})
+```
+
+Above example will show only essential and analytics categories.
+
+#### Policies links
+
+To set policies links mentioned in dialogs use following options:
+
+```
```
Comment (przemyslawzalewski) [inline README.md]: Please toggle on syntax highlighting.

```suggestion
```javascript
```

File: README.md [hunk 15]
```diff
@@ -1,3 +1,95 @@
...
+
+<p align="center"><img width="800" src="images/categories-dialog.png"></a></p>
+
+### Additional configuration options
+
+#### Categories
+
+For showing only specific categories following options needs to be set:
+
+```
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
```javascript
```

File: README.md [hunk 16]
```diff
@@ -1,3 +1,95 @@
...
+    [CC.UNCATEGORIZED]: false,
+    [CC.ESSENTIAL]: true,
+    [CC.PERSONALIZATION]: false,
+    [CC.ANALYTICS]: true,
+    [CC.MARKETING]: false
+  }
+})
+```
+
+Above example will show only essential and analytics categories.
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
The above example will only show essential and analytics categories.
```

File: README.md [hunk 17]
```diff
@@ -1,3 +1,95 @@
...
+
+#### Policies links
+
+To set policies links mentioned in dialogs use following options:
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
To wire the policies links mentioned in dialogs, please use the following options:
```

File: README.md [hunk 18]
```diff
@@ -1,3 +1,95 @@
+[![MIT License][li]][ll]
+
+## Additional instructions for this fork from [Sandstream Development](https://github.com/sandstreamdev)
+
+This fork introduces additional categories dialog which is show after clicking 'Customize settings' button on initial cookies consent dialog. There are some more additional features to handle GDPR - see below.
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
This fork introduces additional categories dialog which is shown after clicking the Customize settings button on the initial cookie consent dialog. There are some more additional features to handle more GDPR concerns - please see below.
```

File: package.json (hunk 1 above)
Comment (marekrozmus) [reply package.json]: Author is reverted to original value.

File: README.md [hunk 19]
```diff
@@ -1,3 +1,95 @@
...
+
+Check [GDPR example](./examples/example-gdpr.html) to see it in action.
+
+<p align="center"><img width="800" src="images/cc-dialog.png"></a></p>
```
Comment (przemyslawzalewski) [inline README.md]: Misclosed anchors?

```suggestion
<p align="center"><img width="800" src="images/cc-dialog.png"></p>
```

File: README.md [hunk 20]
```diff
@@ -1,3 +1,95 @@
...
+
+<p align="center"><img width="800" src="images/categories-dialog.png"></a></p>
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
<p align="center"><img width="800" src="images/categories-dialog.png"></p>
```

File: README.md [hunk 21]
```diff
@@ -1,3 +1,95 @@
...
+
+### Additional configuration options
+
+#### Categories
+
+For showing only specific categories following options needs to be set:
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
For showing only specific categories, the following options need to be set:
```

File: README.md (hunk 19 above)
Comment (marekrozmus) [reply README.md]: nice catch :) copy&paste issue ;)

File: package.json (hunk 3 above)
Comment (marekrozmus) [reply package.json:2]: will rebuild it and update dependencies

File: README.md [hunk 22]
```diff
@@ -1,3 +1,117 @@
...
+#### Module
+```javascript
+import { CookieConsent } from '@sandstreamdev/cookieconsent';
+// or
+const { CookieConsent } = require('@sandstreamdev/cookieconsent');
+```
+
+#### Classic
+```javascript
+const CC = window.CookieConsent
```
Comment (przemyslawzalewski) [inline README.md]:
```suggestion
const CC = window.CookieConsent;
```

**Review by przemyslawzalewski — APPROVED** [review]

**Pull request merged**